.github/workflows/wheels-pure.yml

```diff
@@ -0,0 +1,132 @@
+name: Build RAPIDS project wheels
+
+on:
+  workflow_call:
+    inputs:
+      package-name:
+        required: true
+        type: string
+      package-dir:
+        required: true
+        type: string
+      python-package-cuda-suffix:
+        required: false
+        type: string
+        default: ''
+      python-package-versioneer-override:
+        required: true
+        type: string
+      python-package-build-tag:
+        required: false
+        type: string
+        default: ''
+      test-checkout-src:
+        required: false
+        type: string
+        default: 'true'
+      test-extras:
+        required: false
+        type: string
+        default: ''
+      test-unittest:
+        required: false
+        type: string
+        default: ''
+      test-before:
+        required: false
+        type: string
+        default: 'true'
+
+jobs:
+  wheel-build:
+    name: wheel build pure
+    runs-on: [self-hosted, linux, amd64, cpu4]
+    container:
+      image: "rapidsai/cibuildwheel_x86_64"
+      env:
+        RAPIDS_PY_WHEEL_CUDA_SUFFIX: ${{ inputs.python-package-cuda-suffix }}
+        RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE: ${{ inputs.python-package-versioneer-override }}
+
+    steps:
+    - name: checkout code repo
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # unshallow fetch for setuptools-scm
+        persist-credentials: false
+
+    - name: Build pure python wheel with pip wheel
+      run: python-cp38 -m pip wheel -w ./dist ${{ inputs.package-dir }} --no-deps
+
+    - name: Upload wheels to downloads.rapids.ai
+      run: rapids-upload-wheels-to-s3 dist
+      env:
+        RAPIDS_PY_WHEEL_NAME: ${{ inputs.package-name }}
+        RAPIDS_BUILD_TYPE: "pull-request"
+        RAPIDS_REPOSITORY: ${{ github.repository }}
+        RAPIDS_SHA: ${{ github.sha }}
+        RAPIDS_REF_NAME: ${{ github.ref_name }}
+        RAPIDS_PY_VERSION: "3.8"
+        AWS_ACCESS_KEY_ID: ${{ secrets.RAPIDSAI_GHA_AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.RAPIDSAI_GHA_AWS_SECRET_ACCESS_KEY }}
+
+  wheel-test:
+    name: wheel test pure
+    runs-on: [self-hosted, linux, amd64, gpu-v100-495-1]
+    needs: wheel-build
+    container:
+      image: "rapidsai/citestwheel_x86_64"
```

Comment:
Merge cibuildwheel/citestwheel containers (they're mostly the same)

Reply:
It takes ~1 minute to download the runtime cibw launcher, and ~3-4 minutes to download the devel citestwheel tester - keeping them separate is fine.